packages/components/src/submenu-popover/index.tsx

```diff
@@ -0,0 +1,122 @@
+import { Popover } from '@wordpress/components';
+import { LegacyRef, useCallback, useMemo, useRef, useState } from 'react';
+
+interface SubmenuPopoverProps extends Popover.Props {
+	isVisible?: boolean;
+	offset?: number;
+	placement?:
+		| 'top'
+		| 'top-start'
+		| 'top-end'
+		| 'bottom'
+		| 'bottom-start'
+		| 'bottom-end'
+		| 'right'
+		| 'right-start'
+		| 'right-end'
+		| 'left'
+		| 'left-start'
+		| 'left-end';
+	__unstableForcePosition?: boolean;
+}
+
+/**
+ * Adds a11y support to the submenu popover.
+ * - Closes the popover when pressing Escape.
+ * - Closes the popover when pressing Tab and the focus is on the last element.
+ */
+function useCloseSubmenuA11y() {
+	return useCallback(
+		( {
+			event,
+			lastChild,
+			setIsVisible,
+		}: {
+			event: React.KeyboardEvent< HTMLElement >;
+			lastChild: Element | null | undefined;
+			setIsVisible: ( isVisible: boolean ) => void;
+		} ) => {
+			const isEscape = event.key === 'Escape';
+			const tabOnLastChild = event.key === 'Tab' && ! event.shiftKey && lastChild === event.target;
+			if ( isEscape || tabOnLastChild ) {
+				setIsVisible( false );
+			}
+		},
+		[]
+	);
+}
+
+/**
+ * Checks if the submenu popover has enough space to be displayed on the right.
+ * If not, it will return false to be displayed on the left.
+ */
+function useHasRightSpace( element: HTMLElement | undefined ): boolean {
+	return useMemo( () => {
+		if ( ! element ) {
+			return true;
+		}
+		const calculatedThreshold = element.offsetWidth * 1.1;
+		const { right } = element.getBoundingClientRect();
+		return window.innerWidth - right > calculatedThreshold;
+	}, [ element ] );
+}
+
+/**
+ * Returns the anchor rect adding a given offsetTop.
+ */
+function useAnchorWithOffset( anchorRect: DOMRect | undefined, { offsetTop = 0 } = {} ) {
+	if ( ! anchorRect || offsetTop === 0 ) {
+		return anchorRect;
+	}
+	return new DOMRect( anchorRect.x, anchorRect.y + offsetTop, anchorRect.width, anchorRect.height );
+}
+
+export function useSubenuPopoverProps< T extends HTMLElement >(
```

Comment:
It looks like a typo here, should it be `useSubmenuPopoverProps`?

Reply:
Fixed here: https://github.com/Automattic/wp-calypso/pull/73356/commits/35a63b1b9f8f0a2dd531234c6faeaf462321bd76
Thank you! 🙏